lib/bno080/PinInterface.h

```diff
@@ -0,0 +1,37 @@
+/*
+    SlimeVR Code is placed under the MIT license
+    Copyright (c) 2024 Eiren Rain & SlimeVR contributors
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+#ifndef _H_PIN_INTERFACE_
+#define _H_PIN_INTERFACE_
+
+#include <Arduino.h>
+
+class PinInterface
+{
+public:
+    virtual int digitalRead();
+    virtual void pinMode(uint8_t mode);
+    virtual void digitalWrite(uint8_t val);
```

Comment:
```suggestion
    virtual int digitalRead() = 0;
    virtual void pinMode(uint8_t mode) = 0;
    virtual void digitalWrite(uint8_t val) = 0;
```
I don't see any direct implementation for these, they should be pure virtual then.